controllers/front/ExpressCheckout.php

```diff
@@ -189,7 +189,16 @@ private function createCustomer($email, $firstName, $lastName)
         $customer->email = $email;
         $customer->firstname = $firstName;
         $customer->lastname = $lastName;
-        $customer->passwd = md5(time() . _COOKIE_KEY_);
+
+        if (class_exists('PrestaShop\PrestaShop\Core\Crypto\Hashing')) {
+            $crypto = new PrestaShop\PrestaShop\Core\Crypto\Hashing();
+            $customer->passwd = $crypto->hash(
+                md5(time() . _COOKIE_KEY_),
+                _COOKIE_KEY_
+            );
+        } else {
+            $customer->passwd = md5(time() . _COOKIE_KEY_);
```

Comment:
And in PS passwords are hashed like this md5(_COOKIE_KEY_ . $passwd)   or maybe it doesn't matter :thinking: ?